### PR TITLE
[IMP] Manufacturing: Improved some usability issues.

### PR DESCRIPTION
**Pull request by atp-odoo**
Task : https://www.odoo.com/web#id=10541&view_type=form&model=project.task&menu_id=3942&action=327
pad link : https://pad.odoo.com/p/r.fZCshEnjNImVXxFn

### CODE REVIEW

File: addons/mrp/mrp.py [hunk 1]
```diff
@@ -353,6 +353,14 @@ def onchange_product_tmpl_id(self, cr, uid, ids, product_tmpl_id, product_qty=0,
             }
         return res
 
+    def onchange_product_id(self, cr, uid, ids, product_id, context=None):
+        res = {}
+        if product_id:
+            res['value'] = {
+                'product_tmpl_id': self.pool["product.product"].browse(cr, uid, product_id, context).product_tmpl_id.id,
+            }
+        return res
```
Comment (rim-odoo) [inline addons/mrp/mrp.py]: I suppose you're trying to update the product template of the BoM when created from Manufacturing Order. It seems it does not work. Why don't you have a hidden field on the manufacturing order view with the product template and pass it in as context for new BoMs?

File: addons/mrp/mrp.py [hunk 2]
```diff
@@ -852,7 +860,7 @@ def _calculate_qty(self, cr, uid, production, product_qty=0.0, context=None):
         production_qty = uom_obj._compute_qty(cr, uid, production.product_uom.id, production.product_qty, production.product_id.uom_id.id)
 
         scheduled_qty = {}
-        for scheduled in production.product_lines:
+        for scheduled in production.move_lines:
```
Comment (rim-odoo) [inline addons/mrp/mrp.py]: why this change?

File: addons/mrp/mrp_view.xml [hunk 3]
```diff
@@ -710,8 +715,9 @@
                     </group>
                     <notebook>
                         <page string="Consumed Products">
-                            <group>
-                                <group string="Products to Consume">
+                            <div class="row">
+                                <div class="col-xs-6">
```
Comment (rim-odoo) [inline addons/mrp/mrp_view.xml]: yes, please, it will be done in a more large refacturing of the web client

File: addons/mrp/mrp.py (hunk 2 above)
Comment (bba-odoo) [inline addons/mrp/mrp.py]: @rim-odoo 
manually added stock move in "Products to Consume" field is not considered/added in "Produce" wizard, only the bom components are added on wizard("To Consume" field), check this video: http://youtu.be/XHcRaYinjJ0

File: addons/mrp/mrp_view.xml (hunk 3 above)
Comment (bba-odoo) [inline addons/mrp/mrp_view.xml]: @rim-odoo : i remove the changes :)

File: addons/mrp/mrp.py (hunk 1 above)
Comment (bba-odoo) [inline addons/mrp/mrp.py]: @rim-odoo  : i added hidden field and pass it in context.
Thanks.

File: addons/stock/stock.py [hunk 4]
```diff
@@ -2218,7 +2218,7 @@ def action_assign(self, cr, uid, ids, context=None):
         todo_moves = []
         operations = set()
         for move in self.browse(cr, uid, ids, context=context):
-            if move.state not in ('confirmed', 'waiting', 'assigned'):
+            if move.state not in ('confirmed', 'waiting', 'assigned') and  move.product_id.type != 'consu':
```
Comment (rim-odoo) [inline addons/stock/stock.py]: @bba-odoo Are you sure it won't break "check availability" in stock module?

File: addons/mrp/mrp.py [hunk 5]
```diff
@@ -852,7 +854,7 @@ def _calculate_qty(self, cr, uid, production, product_qty=0.0, context=None):
...
```
Comment (rim-odoo) [inline addons/mrp/mrp.py]: Not sure about that

**Comment by mart-e** [comment]
Something is wrong with this PR, are you sure of your source/target branches ?

**Comment by rim-odoo** [comment]
@mart-e @bba-odoo Just rebase to fix it
